### DISCUSSION TITLE
Idea for add some fallback feature for custom cache

### DISCUSSION THREAD

**Discussion by BruceRodrigues** [vercel/swr#1189] [Ideas]
Hey all,

Just want to give some feedback/idea about the new incoming custom cache provider feature.
I was trying to build a custom cache to add some initial data for every page. That way, if for any reason an API request doesn't work I can fill my components/pages with something, and also don't need to be always worrying about null checks. But to do that I needed to build some fallback feature, so if I make a request using the key `/api/example/1` I can fallback to `/api/example/dummy`.

I think it would be awesome if we had something like that out-of-the-box. 
Also, since you're making cache more flexible, I think it would be a good idea if we had an `unserializable` function to get the key's original value.

Let me know what you guys think!

**Comment by huozhi** [comment]
@BruceRodrigues Thanks for these great feedback! Do you have any usage examples for `unserializable` cases you're trying to work with? That would be very helpful for us to improve the cache API
